src/main/java/org/jabref/logic/importer/fileformat/ACMPortalParser.java

```diff
@@ -0,0 +1,254 @@
+        }
+        return bibEntries;
+    }
+
+    /**
+     * Parse all DOIs from the ACM Portal search results page
+     *
+     * @param stream html stream
+     * @return DOI list
+     */
+    public List<String> parseDoiSearchPage(InputStream stream) throws ParseException {
+        List<String> doiList = new ArrayList<>();
+        String htmlLine;
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8))) {
+            int cnt = 0;
+            while ((htmlLine = in.readLine()) != null && cnt < MAX_ITEM_CNT_PER_PAGE) {
+                if (ITEM_HTML.equals(htmlLine)) {
+                    Matcher matcher = DIO_HTML_PATTERN.matcher(in.readLine());
+                    if (matcher.find()) {
+                        doiList.add(matcher.group(1));
+                        ++cnt;
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new ParseException(e);
+        }
+        return doiList;
+    }
+
+    /**
+     * Obtain BibEntry according to DOI
+     *
+     * @param doiList DOI List
+     * @return BibEntry List
+     */
+    public List<BibEntry> getBibEntriesFromDoiList(List<String> doiList) throws FetcherException {
+        List<BibEntry> bibEntries = new ArrayList<>();
+        CookieHandler.setDefault(new CookieManager());
+        try (InputStream stream = new URLDownload(getUrlFromDoiList(doiList)).asInputStream()) {
+            String jsonString = new String((stream.readAllBytes()), Charsets.UTF_8);
+
+            JsonElement jsonElement = JsonParser.parseString(jsonString);
+            if (jsonElement.isJsonObject()) {
+                JsonArray items = jsonElement.getAsJsonObject().getAsJsonArray("items");
+                for (JsonElement item : items) {
+                    for (Map.Entry<String, JsonElement> entry : item.getAsJsonObject().entrySet()) {
+                        bibEntries.add(parseBibEntry(entry.getValue().toString()));
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            throw new FetcherException("A network error occurred while fetching from ", e);
+        }
+
+        return bibEntries;
+    }
+
+    /**
+     * Constructing the query url for the doi
+     *
+     * @param doiList DOI List
+     * @return query URL
+     */
+    public URL getUrlFromDoiList(List<String> doiList) throws URISyntaxException, MalformedURLException {
+        URIBuilder uriBuilder = new URIBuilder(DOI_URL);
+        uriBuilder.addParameter("targetFile", "custom-bibtex");
+        uriBuilder.addParameter("format", "bibTex");
+        uriBuilder.addParameter("dois", String.join(",", doiList));
+        return uriBuilder.build().toURL();
+    }
+
+    private StandardEntryType typeStrToEnum(String typeStr) {
+        StandardEntryType type;
+        typeStr = typeStr.toLowerCase(Locale.ENGLISH).replace("_", "");
```

Comment:
I think you can simplify this whole EntryType stuff, if you want too look up an enum key by the string value:
StandardEntryType.valueOf(...) 
Or alternatively, use the guava equivalent https://guava.dev/releases/19.0/api/docs/index.html?com/google/common/base/Enums.html

Reply:
The problem here is that the entry of StandardEntryType uses big camelcase nomenclature, I can't directly use StandardEntryType.valueOf(...), or I need to find a good way from JabRef to deal with big camelcase format and all-caps dashed separation format conversion.

And some types of naming do not match exactly. For example, the API returns PAPER_CONFERENCE corresponding to the CONFERENCE of StandardEntryType.

Reply:
Ah I understand, however, you can directly operate on the Enum, there is the possibility to use an EnumSet and you have an Optional, so you don't need to create a new Map with the keys and values 
```java
       Optional<StandardEntryType> foundType  = EnumSet.allOf(StandardEntryType.class).stream().filter(type-> type.getDisplayName().equals(strType)).findAny(); 
```

Reply:
>   ```java
> Optional<StandardEntryType> foundType  = EnumSet.allOf(StandardEntryType.class).stream().filter(type-> type.getDisplayName().equals(strType)).findAny(); 
> ```
This is invalid if the input string is ARTICLE


New version~
```java
private StandardEntryType typeStrToEnum(String typeStr) {
    StandardEntryType type;
    if ("PAPER_CONFERENCE".equals(typeStr)) {
        type = StandardEntryType.Conference;
    } else {
        String upperUnderscoreTyeStr = CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, typeStr);
        type = Enums.getIfPresent(StandardEntryType.class, upperUnderscoreTyeStr).or(StandardEntryType.Article);
    }
    return type;
}
```